src/megachatapi_impl.cpp

```diff
@@ -5282,10 +5328,29 @@ MegaChatSessionHandler::~MegaChatSessionHandler()
 
 void MegaChatSessionHandler::onSessStateChange(uint8_t newState)
 {
-    if (newState == rtcModule::ISession::kStateInProgress)
+    MegaChatCallPrivate* chatCall = callHandler->getMegaChatCall();
+
+    assert(chatCall != NULL);
+    if (chatCall != NULL)
     {
-        MegaChatCallPrivate* chatCall = callHandler->getMegaChatCall();
-        chatCall->setRemoteAudioVideoFlags(session->receivedAv());
+        if (newState == rtcModule::ISession::kStateInProgress)
+        {
+            chatCall->setRemoteAudioVideoFlags(session->receivedAv());
+
+            if (chatCall->getRemoteStatus() == rtcModule::ICall::kStateRingIn)
```

Comment:
What's the purpose of this condition? in case the status is different, should we `assert` or log a warning?